.github/workflows/WorkitemValidation.yaml

```diff
@@ -15,16 +15,46 @@ defaults:
 
 jobs:
   WorkItemValidation:
+    name: 'Validate link to issues'
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
+      - name: Link work items to pull request if possible
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          build/scripts/PullRequestValidation/LinkPullRequestToWorkItem.ps1 -PullRequestNumber ${{ github.event.pull_request.number }} -Repository ${{ github.repository }}
+
       - name: Validate work items for pull request
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           build/scripts/PullRequestValidation/ValidateIssuesForPullRequest.ps1 -PullRequestNumber ${{ github.event.pull_request.number }} -Repository ${{ github.repository }}
+  
+  WorkitemValidationForMicrosoft:
+    name: 'For Microsoft: Validate link to internal workitems' 
+    runs-on: ubuntu-latest
+    needs: WorkItemValidation
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Validate internal work items for pull request
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          build/scripts/PullRequestValidation/ValidateInternalWorkitemForPullRequest.ps1 -PullRequestNumber ${{ github.event.pull_request.number }} -Repository ${{ github.repository }}
+
+
+  AddMilestoneToPullRequest:
```

Comment:
Why a separate job for the milestone? Could just be a last step in the previous job. Or am I missing your point?

Reply:
We can.. The idea behind splitting them was that the previous job will most likely become a required check. I didn't think we should block PRs from merging if linking the PR to a milestone failed for some reason.

Reply:
If we are to trust milestones, then PRs needs to be linked to milestones always. 

The logic isn't that complex, so I don't expect to have it failing regularly.